flow/genericactors.actor.h

```diff
@@ -776,13 +776,20 @@ Future<Void> setAfter( Reference<AsyncVar<T>> var, double time, T val ) {
 }
 
 ACTOR template <class T>
-Future<Void> resetAfter( Reference<AsyncVar<T>> var, double time, T val ) {
+Future<Void> resetAfter( Reference<AsyncVar<T>> var, double time, T val, int warningLimit = -1, const char* context = NULL ) {
 	state bool isEqual = var->get() == val;
 	state Future<Void> resetDelay = isEqual ? Never() : delay(time);
+	state int resetCount = 0;
+	state double lastReset = now();
 	loop {
 		choose {
 			when( wait( resetDelay ) ) {
 				var->set( val );
+				resetCount++;
+				if(context && warningLimit >= 0 && resetCount > warningLimit) {
+					TraceEvent(SevWarnAlways, context).detail("ResetCount", resetCount).detail("LastReset", now() - lastReset);
```

Comment:
I guess it's hard to say without seeing how frequently we enter degraded states, but with processes that are pretty long-lived we may see this happening spuriously. We may want to keep an eye on this and possibly limit reporting to times when the gap between `lastReset` isn't too large.

Reply:
I will add time limit that will reset restCount